main/uart_nic.c

```diff
@@ -403,37 +514,115 @@ static void IRAM_ATTR check_online_status() {
     }
 }
 
+static void IRAM_ATTR clear_stored_ssids() {
+    memset(scan.stored_ssids, 0, scan.stored_ssids_count * sizeof(ScanResult));
+    scan.stored_ssids_count = 0;
+}
+
+static void IRAM_ATTR store_scanned_ssids(wifi_ap_record_t *aps, int ap_count) {
+    for (uint8_t i = 0; i < ap_count; ++i) {
+        bool found = false;
+        if (scan.stored_ssids_count == SCAN_MAX_STORED_SSIDS) {
+            break;
+        }
+        for (uint8_t j = 0; j < scan.stored_ssids_count; ++j) {
+            ESP_LOGI(TAG, "Comparing >%s< and %s", (char *)scan.stored_ssids[j].ssid, (char *)aps[i].ssid);
+            if (memcmp(scan.stored_ssids[j].ssid, aps[i].ssid, SSID_LEN) == 0) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            memcpy(scan.stored_ssids[scan.stored_ssids_count].ssid, aps[i].ssid, SSID_LEN);
```

Comment:
:thinking: Is memcpy the right one? Is aps[i].ssid also an array of at least such size, or is it just a null-terminated string and the valid memory might be shorter?

Reply:
Then maybe a static_assert checking that both things have the same size, eg `static_assert(sizeof(stored_ssids.ssid) == sizeof(aps.ssid) && == SSID_LEN)`? Just to make sure nothing breaks terribly if we ever update the ESP SDK or something.

Reply:
Sadly I can't use `static_assert` in the c codebase. We should be able to use c++ and might be a good idea to investigate it in the future.